modules/analysis/src/service/load.rs

```diff
@@ -236,7 +240,22 @@ impl AnalysisService {
                 .distinct()
                 .into_query(),
             GraphQuery::Query(query) => sbom_node::Entity::find()
-                .filtering(query.clone())?
+                // TODO: The following translation effectively ignores
+                // any fields named 'purl' or 'cpe', potentially
+                // selecting every SbomId in the db, without any other
+                // fields in the query. We need help constructing the
+                // SELECT statement that would enable us to find
+                // partial matches of the purl's stored within
+                // entity::qualified_purl::Column::Purl, i.e. the
+                // 'PURL' column of the 'QUALIFIED_PURL' table.
```

Comment:
@JimFuller-RedHat help?

Reply:
maybe something like

`SELECT * FROM qualified_purl WHERE get_purl(id)::text LIKE '%test%';`

Reply:
I think 'we got the graph side covered' does the needful with respect to this bug ... the formal definition of url and uri, order of url params are not significant.

Reply:
I'm sorry, I don't understand.

Reply:
Maybe we just compare the versioned purl, i.e. all but the qualifiers, in the load? And then provide extra logic in the graph query to account for the qualifier order.

Reply:
But that won't pick up partial matches in the qualifiers, e.g. `purl~curl` would match but `purl~i386` would not.

Reply:
I am not proposing fixing filter queries for partial match on purl in postgres db with this PR ... sounds more complicated then just fixing analysis graph (though admittedly not as satisfying!).

Reply:
the one where people use 'api/v2/analysis/component'

Reply:
No, I meant what are the uses of that URL that need to work and currently don't? I think I'm asking for an answer to [this question you posed](https://github.com/trustification/trustify/issues/1280#issuecomment-2685100672). I think I can make the globbing work (alternative 1) using your suggestion of `get_purl` with caveats, in particular that a glob that includes some portion of a versioned purl *AND* one or more qualifiers may return a false negative.

You have a better idea of what globs users will search for than me to know whether that's acceptable.